File: src/main/resources/static/user/angular-js/service/sale-off-service.js
solar_app.service('SaleOffService', function ($http) {

    this.findAll = function () {
        return $http({
            method: 'GET',
            url: API_SaleOff,
        })
    }

    this.findByProductId = function (productId) {
        return $http({
            method: 'GET',
            url: API_SaleOff + '/' + productId,
        })
    }
});

File: src/main/java/com/main/service/SaleOffService.java
package com.main.service;


import com.main.entity.SaleOff;

import java.util.List;

public interface SaleOffService {
    List<SaleOff> findAll();

    SaleOff findByProductId(String id);

    SaleOff findById(Integer id);

    void save(SaleOff saleOff);

    boolean doseExitsProductId(String id);

    void updateSalsOffStatus();

    void delete(SaleOff saleOff);
}
